src/js_native_api_types.h

```diff
@@ -115,4 +115,23 @@ typedef struct {
   napi_status error_code;
 } napi_extended_error_info;
 
+typedef enum {
```

Comment:
The new enums need to be guarded by `#ifdef NAPI_EXPERIMENTAL`.

Reply:
if we define new enums to `NAPI_EXPERIMENTAL`, what about the `napi_get_property_names` 🤔

Reply:
We have [`NAPI_EXPERIMENTAL` definition](https://github.com/nodejs/node/blob/master/src/js_native_api_v8.cc#L4) in Node.js build. Addons using N-API would not be affected whether or not we use experimental definitions in Node.js implementation for .cc files not being included in their source codes.